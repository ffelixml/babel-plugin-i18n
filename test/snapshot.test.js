const { join, basename } = require('path');
const { readdirSync, readFileSync } = require('fs');

const testPlugin = require('./testPlugin');

const FIXTURE_PATH = join(__dirname, 'fixtures');

const testFiles = readdirSync(FIXTURE_PATH).filter(file => (
  file.match(/\.js$/)
));

const testNames = testFiles.map(file => basename(file, '.js'));
const translationJSON = readFileSync(join(FIXTURE_PATH, 'translations.json'), 'utf8');
const translations = JSON.parse(translationJSON);

testNames.forEach((name) => {
  test(name, () => {
    const source = readFileSync(join(FIXTURE_PATH, `${name}.js`), 'utf8');
    const result = testPlugin(source, translations);
    expect(result).toMatchSnapshot();
  });
});
