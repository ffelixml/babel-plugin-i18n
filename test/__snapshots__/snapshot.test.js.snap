exports[`test objects 1`] = `
"const categories = {
  party: \'Party\',
  wedding: \'Wedding\'
};
console.log(categories.party);

const otherDataTypes = {
  boolean: false,
  null: null,
  number: 42,
  array: [1, 2, 3]
};
console.log(otherDataTypes);"
`;

exports[`test strings 1`] = `
"\'Hello, world!\';
\'Two levels deep translation\';
\'Very nested object\';

const numberOne = 1;
const booleanTrue = true;
const thisIsNull = null;"
`;
